cpp/util/fake_etcd.cc

```diff
@@ -178,26 +181,37 @@ void FakeEtcdClient::InternalPut(const string& key, const string& value,
 }
 
 
+void FakeEtcdClient::UpdateOperationStats(const string& op, const Task* task) {
+  CHECK_NOTNULL(task);
+  if (!task->IsActive()) {
+    std::lock_guard<std::mutex> lock(mutex_);
+    ++stats_[op + (task->status().ok() ? "Success" : "Fail")];
+  }
+}
+
+
 void FakeEtcdClient::Create(const string& key, const string& value,
                             Response* resp, Task* task) {
-  return InternalPut(key, value, system_clock::time_point::max(), true, -1,
-                     resp, task);
+  InternalPut(key, value, system_clock::time_point::max(), true, -1, resp,
+              task);
+  UpdateOperationStats("create", task);
```

Comment:
This isn't safe, the ```task``` could have already been deleted... You could register it as a cleanup callback on the task, though, those get run after ```Return``` has been called, but before the task is gone, so you could check the status on it. Plus, you ensure it getting called, no matter what crazy code path is taken.